lib/ebay_api/operations/sell/inventory/offers/list.rb

```diff
@@ -0,0 +1,19 @@
+class EbayAPI
+  scope :sell do
+    scope :inventory do
+      scope :offers do
+        operation :list do
+          option :sku
+          option :limit, optional: true
+          option :offset, optional: true
+
+          http_method :get
+          query do
+            { sku: sku, limit: limit, offset: offset }.select { |_, v| v }
+          end
+          response(404) { nil }
```

Comment:
Why `nil` again? Isn't something `Enumerable` expected to be returned (empty array, for example)?

Reply:
I agree that `nil` is out of place here.

I see a similar case [here](https://github.com/nepalez/ebay_api/blob/95a76ef46213f3672a011b5f8b16243638a58f0d/lib/ebay_api/operations/sell/marketing/ads/list.rb#L21). In that case eBay documentation says that 404 is returned "the campaign ID submitted does not exist" (BTW ebay_api does not support the required `campaign_id` option) and ebay_api raises an exception.

Here in the offer list case, a 404 is returned when there is no offer for the specified SKU, even if the item with this SKU exists - this is what I found by trial and error.

Question is, should ebay_api return `[]` in both cases or should there be an exception for ads list.

I'm sure `[]` is appropriate for offer list - eBay is actually doing the wrong thing by sending 404 just because there are no offers for specified SKU.

Reply:
Actually in case of promoted listings the required option is specified for entire ads scope here: https://github.com/nepalez/ebay_api/blob/95a76ef46213f3672a011b5f8b16243638a58f0d/lib/ebay_api/operations/sell/marketing/ads.rb

This is problem of evil-client based libs that to understand behavior and requirements for operation you need to look in all nested scopes and merge definitions in mind.